pkg/internal/infrastructure/template.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package infrastructure
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+	"text/template"
+
+	"github.com/Masterminds/sprig"
+	"github.com/pkg/errors"
+)
+
+var (
+	//go:embed templates/main.tpl.tf
+	mainFile string
+	//go:embed templates/terraform.tfvars
+	terraformTFVars []byte
+	//go:embed templates/variables.tf
+	variablesTF string
+
+	mainTemplate *template.Template
+)
+
+func init() {
+	var err error
+	mainTemplate, err = template.
+		New("main.tf").
+		Funcs(sprig.TxtFuncMap()).
+		Funcs(
+			map[string]interface{}{
+				"required": required,
+			}).
+		Parse(mainFile)
+
+	if err != nil {
+		panic(err)
+	}
+}
+
+func required(message string, val interface{}) (interface{}, error) {
```

Comment:
It's not necessary but I consider it a plus for the development more than the actual runtime checks. 

As a concrete example, doing away with helm means we no longer have the `values.yaml` and default values.  I already had to debug some missing values for the template e.g. `platform_update_domain_count` output keys where the template was rendering with no complaints. Previously it was was working due to the defaults in `values.yaml`. Would that be a required value it would be easier to identify that it was missing.

I agree that the value outside "assertions during development" is limited and if you wish it can be removed.

Reply:
I do not vote for or against it. I guess it is not required as the values are now always generated by the extension and there is no way to do `helm template` anymore, which was actually the primary reason for `required`.
But as said it does not harm.

Reply:
@rfranzke @dkistner thank you very much for your input. I decided in the last commit to remove the required function. Let's start simple and if there is a need for it we can always introduce it back.